### PR TITLE
Fix PS-1827 (LP #1736921: Inconsistent and unsafe FLUSH behavior in terms of replication) (8.0)

### PR DESCRIPTION
**Pull request by percona-ysorokin**

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Jenkins build link
https://ps2.cd.percona.com/view/8.0/job/percona-server-8.0-param/19/

`binlog_nogtid.binlog_persist_only_variables`, `binlog_nogtid.binlog_persist_variables` and `main.all_persisted_variables` already fixed and re-recorded.

File: mysql-test/r/all_persisted_variables.result [hunk 1]
```diff
@@ -52,9 +52,9 @@ include/assert.inc ['Expect 413 persisted variables in the table. Due to open Bu
 ************************************************************
 # restart
 
-include/assert.inc ['Expect 353 persisted variables in persisted_variables table.']
-include/assert.inc ['Expect 353 persisted variables shown as PERSISTED in variables_info table.']
-include/assert.inc ['Expect 353 persisted variables with matching peristed and global values.']
+include/assert.inc ['Expect 410 persisted variables in persisted_variables table.']
```
Comment (laurynas-biveinis) [inline mysql-test/r/all_persisted_variables.result:55]: 60 variable difference?!

File: mysql-test/suite/binlog_gtid/t/percona_binlog_skip_flush_commands.test [hunk 2]
```diff
@@ -0,0 +1,204 @@
+--source include/have_log_bin.inc
```
Comment (laurynas-biveinis) [inline mysql-test/suite/binlog_gtid/t/percona_binlog_skip_flush_commands.test:1]: Ideally there could be a binlog_nogtid test too that drops the check for gtids, but I am not sure whether effort/payoff ratio is good enough for that

File: sql/mysqld.cc [hunk 3]
```diff
@@ -1036,6 +1036,7 @@ bool binlog_gtid_simple_recovery;
 ulong binlog_error_action;
 const char *binlog_error_action_list[] = {"IGNORE_ERROR", "ABORT_SERVER",
                                           NullS};
+bool opt_binlog_skip_flush_commands = 0;
```
Comment (laurynas-biveinis) [inline sql/mysqld.cc]: !!!

```suggestion
bool opt_binlog_skip_flush_commands = false;
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: mysql-test/r/all_persisted_variables.result (hunk 1 above)
Comment (percona-ysorokin) [reply mysql-test/r/all_persisted_variables.result:55]: Here, assert **message** hasn't been updated for a long time. For some reason, upon adding a new var, developers were updating only assert **condition**. Before this commit the message should have been set to 409, but it wasn't.

File: sql/mysqld.cc (hunk 3 above)
Comment (percona-ysorokin) [reply sql/mysqld.cc]: Fixed

**Comment by percona-ysorokin** [comment]
Updated Jenkins build link with selected tests
https://ps2.cd.percona.com/view/8.0/job/percona-server-8.0-param/23/

**Pull request merged**